### PR TITLE
Improve API error handling

### PR DESCRIPTION
**Pull request by flosse**
- [x] Define the JSON shape of  an API error
- [ ] Create an error struct that implements the `Responder` trait
- [ ] Write / adjust tests that checks known error responses
- [ ] Implement the `From` trait for the new error struct

### CODE REVIEW

**Comment by uklotzde** [comment]
### Inspiration
Rejection handler for generating uniform error responses (warp + thiserror/anyhow):
https://gitlab.com/uklotzde/aoide-rs/-/blob/development/src/api/web/mod.rs#L128

File: ofdb-boundary/src/lib.rs [hunk 1]
```diff
@@ -1213,3 +1213,12 @@ impl From<Entry> for UpdatePlace {
         }
     }
 }
+
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(feature = "extra-derive", derive(Debug, Clone, PartialEq))]
+pub struct Error {
+    /// HTTP status code
+    pub status: u16,
```
Comment (flosse) [inline ofdb-boundary/src/lib.rs:1221]: probably better `http_status`

**Comment by flosse** [comment]
BTW: https://httpstatuses.com/

**Comment by flosse** [comment]
This is mostly done and probably will be different when we switch to warp.

**Pull request closed without merging**